src/pybind/mgr/dashboard/frontend/src/app/shared/directives/form-input-disable.directive.ts

```diff
@@ -0,0 +1,36 @@
+import { AfterViewInit, Directive, ElementRef, Input, Optional } from '@angular/core';
+
+import { Permissions } from '../models/permissions';
+import { AuthStorageService } from '../services/auth-storage.service';
+import { FormNameDirective } from './form-name.directive';
+
+@Directive({
+  selector:
+    'input:not([cdNoFormInputDisable]), select:not([cdNoFormInputDisable]), [cdFormInputDisable]'
```

Comment:
Currently we are applying this to all form `input`s and `select`s.
We should only apply it on forms that have the `cdFormName `attribute or inputs/selects that have `cdFormInputDisable `attribute:
```suggestion
    '[cdFormName] input, [cdFormName] select, [cdFormInputDisable]'
```

Reply:
I tried to specify the attribute like this but its not working as intended. According to this article, [(https://medium.com/swlh/anatomy-of-angular-attribute-directive-selector-a1d83a73242) if I give the selector as this, it will only gets applied the `input` field with `cdFormName` attribute. @tspmelo

Reply:
@tspmelo Angular selectors are very limited: they only supports elements, classes, attributes and the pseudo-class `:not()`:
```ts
// ngForm directive
selector: 'form:not([ngNoForm]):not([formGroup]),ng-form,[ngForm]'
```

As far as I can see, there are only 2 ways of doing this:
* `FormName` parent element looks for potentially disable-able children elements: 
  * By means of `ViewChildren` query on `FormInput` elements (`*Children` selector can only be used for [a limited set of elements](https://stackoverflow.com/questions/49162473/what-are-all-the-valid-selectors-for-viewchild-and-contentchild))
  * Through a query on any `ElementRef`, but then we should implement code to cycle through it and discard non `input` elements.
* `FormInput` uses dependency-injector to find the parent scope.

Other approaches involve adding lots of directives to any `input` element... Am I missing anything? Is there any other way to do this without having to manually tag every single input?

Reply:
@tspmelo I ran `ng.profiler.timeChangeDetection()` on some of the pages having the form element including the 3 I've made changes on and I couldn't see it impacting the performance drastically. 

**With the new changes the result in osd-flags-modal component is**
```
ng.profiler.timeChangeDetection()
common_tools.ts:64 ran 961 change detection cycles
common_tools.ts:65 **0.52 ms per check**
ChangeDetectionPerfRecord {msPerTick: 0.5204994797085846, numTicks: 961}
```
**Without the change on it is:**
```
ng.profiler.timeChangeDetection()
common_tools.ts:64 ran 822 change detection cycles
common_tools.ts:65 **0.61 ms per check**
ChangeDetectionPerfRecord {msPerTick: 0.6086922141118306, numTicks: 822}
```
Same goes for the Recovery priority component 
**With the change:**
```
ng.profiler.timeChangeDetection()
common_tools.ts:64 ran 884 change detection cycles
common_tools.ts:65 **0.57 ms per check**
ChangeDetectionPerfRecord {msPerTick: 0.5657861990952189, numTicks: 884}
```

**Without the change:**
```
ng.profiler.timeChangeDetection()
common_tools.ts:64 ran 866 change detection cycles
common_tools.ts:65 **0.58 ms per check**
ChangeDetectionPerfRecord {msPerTick: 0.5780311778291296, numTicks: 866}
```

And for PG Scrub modal
**With the change:**
```
ng.profiler.timeChangeDetection()
common_tools.ts:64 ran 700 change detection cycles
common_tools.ts:65 **0.72 ms per check**
ChangeDetectionPerfRecord {msPerTick: 0.7150857142856486, numTicks: 700}
```

**Without the change:**
```
ng.profiler.timeChangeDetection()
common_tools.ts:64 ran 697 change detection cycles
common_tools.ts:65 **0.72 ms per check**
ChangeDetectionPerfRecord {msPerTick: 0.7185868005735586, numTicks: 697}
```

Thanks @epuertat for this cool debugging tool.

Reply:
@nizamial09 thanks a lot for the detailed profiling! I was a bit concerned about applying directives to basic HTML elements, but it doesn't seem justified.